### PR TITLE
feat: define nexus gateway msgs

### PR DESCRIPTION
**Pull request by fish-sammy**
## Todos

- [ ] Unit tests
- [ ] Manual tests
- [ ] Documentation
- [ ] Connect epics/issues

### CODE REVIEW

File: contracts/nexus-gateway/src/error.rs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+use cosmwasm_std::StdError;
+use thiserror::Error;
+
+use axelar_wasm_std_derive::IntoContractError;
+
+#[derive(Error, Debug, PartialEq, IntoContractError)]
+pub enum ContractError {
+    #[error(transparent)]
+    Std(#[from] StdError),
```
Comment (cgorenflo) [inline contracts/nexus-gateway/src/error.rs:9]: you shouldn't need this

**Review by cjcobb23 — CHANGES_REQUESTED** [review]

File: contracts/nexus-gateway/src/msg.rs [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+#[cw_serde]
+pub struct InstantiateMsg {
+    pub nexus: String,
+    pub router: String,
+}
+
+#[cw_serde]
+pub enum ExecuteMsg {
+    VerifyMessages(Vec<NewMessage>),
+    RouteMessages(Vec<NewMessage>),
```
[the remark this answers is not in the document]
Comment (fish-sammy) [reply contracts/nexus-gateway/src/msg.rs]: Just to keep the interface the same as I put an `unimplemented!` there suggesting there's no intention to implement at all. Don't have a strong opinion though.

**Review by cgorenflo — APPROVED** [review]
can you be a little bit more specific with the PR title/commit?
At least something like `feat: define nexus gateway msgs`

File: contracts/nexus-gateway/src/msg.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (cjcobb23) [reply contracts/nexus-gateway/src/msg.rs]: I think I would prefer to remove it since it's just going to error, but I also don't have a strong opinion

**Review by cjcobb23 — APPROVED** [review]

**Pull request merged**